viewing_party/party.py

```diff
@@ -0,0 +1,253 @@
+            for movie in movie_dict.values():
+                unique_watched_set.add(movie)
+        for friends_movie_dicts in friends_watched_list:
+            for title in friends_movie_dicts.values():
+                friends_watched_set.add(title)
+
+    set_difference = unique_watched_set - friends_watched_set
+    unique_watched_list = list(set_difference)
+
+    users_unique_watched_list = []
+    for unique_movies in unique_watched_list:
+        list_of_dicts = {"title": unique_movies}
+        users_unique_watched_list.append(list_of_dicts)
+
+    return users_unique_watched_list
+
+
+# 2
+
+def get_friends_unique_watched(user_data):
+    user_watched_list = []
+    for user_dicts in user_data["watched"]:
+        user_watched_list.append(user_dicts)
+
+    friends_watched_list = []
+    for friend_list in user_data["friends"]:
+        for friend_dicts in friend_list["watched"]:
+            friends_watched_list.append(friend_dicts)
+
+    user_watched_set = set()
+    friends_watched_set = set()
+    for list_of_dicts in user_data["friends"]:
+        for watched_list in list_of_dicts["watched"]:
+            if len(watched_list) == 0:
+                return []
+        else: 
+            for movie_dict in user_watched_list:
+                for movie in movie_dict.values():
+                    user_watched_set.add(movie)
+            for friends_movie_dicts in friends_watched_list:
+                for title in friends_movie_dicts.values():
+                    friends_watched_set.add(title)
+
+    set_difference = friends_watched_set - user_watched_set
+    unique_watched_list = list(set_difference)
+
+    friends_unique_watched_list = []
+    for unique_movies in unique_watched_list:
+        list_of_dicts = {"title": unique_movies}
+        friends_unique_watched_list.append(list_of_dicts)
+
+    return friends_unique_watched_list
+
+
+# Wave 4 **********************************************
+
+# 1
+
+def get_available_recs(user_data):
+    user_movies = []
+    for movie in user_data["watched"]:
+        user_movies.append(movie["title"])
+
+    recommended_movies = []
+    for friends_watched in user_data["friends"]:
+        for friend_dicts in friends_watched["watched"]:
+            if (
+                friend_dicts["title"] not in user_movies
+                and friend_dicts["host"] in user_data["subscriptions"]
+                and friend_dicts not in recommended_movies
+            ):
+                recommended_movies.append(friend_dicts)
+    
+    return recommended_movies
+
+
+# Wave 5 **********************************************
+
+# 1
+
+def get_new_rec_by_genre(user_data):
+    import statistics
+    from statistics import mode
+
+    favorite_genres = []
+    for watched in user_data["watched"]:
+        favorite_genres.append(watched["genre"])
+    
+    if len(favorite_genres) == 0:
+        return []
+    else:
+        favorite_genre = mode(favorite_genres)
```

Comment:
Don't you already have a function you wrote to calculate the user's favorite genre?